File: app/src/main/java/com/fun/hairclipper/UI/ChangeTrimmerActivity.java
package com.fun.hairclipper.UI;

import android.content.Intent;
import android.media.AudioManager;
import android.os.Bundle;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;

import com.fun.hairclipper.R;
import com.fun.hairclipper.tools.Tool;
import com.google.ads.mediation.admob.AdMobAdapter;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdSize;
import com.google.android.gms.ads.AdView;

public class ChangeTrimmerActivity extends BaseClass {
    private FrameLayout frameLayout;
    private AudioManager audioManager = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_change_trimmer);
        frameLayout = findViewById(R.id.adaptivemain);
        setVolumeControlStream(AudioManager.STREAM_MUSIC);
        setVolumeMax();
        if (!getPaymentSubscription().isPurchased() && Tool.isNetworkAvailable(this)) {
            loadCollapsibleBanner();
        } else {
            frameLayout.setVisibility(View.GONE);
        }
        ImageView home = findViewById(R.id.home_btn);
        home.setOnClickListener(view -> {
            finish();
        });
        ImageView machine1 = findViewById(R.id.machine1);
        machine1.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine1.class);
            startActivity(intent);
        });
        ImageView machine2 = findViewById(R.id.machine2);
        machine2.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine2.class);
            startActivity(intent);
        });
        ImageView machine3 = findViewById(R.id.machine3);
        machine3.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine3.class);
            startActivity(intent);
        });
        ImageView machine4 = findViewById(R.id.machine4);
        machine4.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine4.class);
            startActivity(intent);
        });
        ImageView machine5 = findViewById(R.id.machine5);
        machine5.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine5.class);
            startActivity(intent);
        });
        ImageView machine6 = findViewById(R.id.machine6);
        machine6.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine6.class);
            startActivity(intent);
        });
        ImageView machine7 = findViewById(R.id.machine7);
        machine7.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine7.class);
            startActivity(intent);

        });
        ImageView machine8 = findViewById(R.id.machine8);
        machine8.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine8.class);
            startActivity(intent);

        });
        ImageView machine9 = findViewById(R.id.machine9);
        machine9.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine9.class);
            startActivity(intent);

        });
        ImageView machine10 = findViewById(R.id.machine10);
        machine10.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine10.class);
            startActivity(intent);

        });
        ImageView machine11 = findViewById(R.id.machine11);
        machine11.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine11.class);
            startActivity(intent);

        });
        ImageView machine12 = findViewById(R.id.machine12);
        machine12.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine12.class);
            startActivity(intent);

        });
        ImageView machine13 = findViewById(R.id.machine13);
        machine13.setOnClickListener(view -> {
            Intent intent = new Intent(ChangeTrimmerActivity.this, machine13.class);
            startActivity(intent);
        });
    }

    private void loadCollapsibleBanner() {
        Bundle extras = new Bundle();
        extras.putString("collapsible", "top");
        AdRequest adRequest = new AdRequest.Builder().addNetworkExtrasBundle(AdMobAdapter.class, extras).build();
        AdView adView = new AdView(this);
        adView.setAdUnitId(getString(R.string.CollapsibleBannerAd));
        frameLayout.addView(adView);
        AdSize adSize = getAdSize();
        adView.setAdSize(adSize);
        adView.loadAd(adRequest);
    }

    private void setVolumeMax() {
        audioManager = (AudioManager) getSystemService(AUDIO_SERVICE);
        audioManager.setStreamVolume(AudioManager.STREAM_MUSIC, 1, 0);
    }
}

File: app/src/main/java/com/fun/hairclipper/UI/ChangeTrimmerActivity.kt
package com.`fun`.hairclipper.UI

import android.content.Intent
import android.media.AudioManager
import android.os.Bundle
import android.view.View
import com.`fun`.hairclipper.R
import com.`fun`.hairclipper.admobHelper.AdConstants
import com.`fun`.hairclipper.admobHelper.BannerAd
import com.`fun`.hairclipper.admobHelper.FullScreenAdListener
import com.`fun`.hairclipper.admobHelper.MyApplication
import com.`fun`.hairclipper.admobHelper.RemoteConfig
import com.`fun`.hairclipper.admobHelper.internetConnection
import com.`fun`.hairclipper.databinding.ActivityChangeTrimmerBinding

class ChangeTrimmerActivity : BaseClass() {
    private val binding by lazy { ActivityChangeTrimmerBinding.inflate(layoutInflater) }
    private lateinit var audioManager: AudioManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        audioManager = getSystemService(AUDIO_SERVICE) as AudioManager
        volumeControlStream = AudioManager.STREAM_MUSIC
        setVolumeMax()
        loadCollapsibleBanner()

        binding.homeBtn.setOnClickListener { handleBackPressed() }
        binding.machine1.setOnClickListener {
            startActivity(Intent(this, Machine1::class.java))
        }
        binding.machine2.setOnClickListener { showInterstitialAndNavigate(Machine2::class.java) }
        binding.machine3.setOnClickListener { showInterstitialAndNavigate(Machine3::class.java) }
        binding.machine4.setOnClickListener { showInterstitialAndNavigate(Machine4::class.java) }
        binding.machine5.setOnClickListener { showInterstitialAndNavigate(Machine5::class.java) }
        binding.machine6.setOnClickListener { showInterstitialAndNavigate(Machine6::class.java) }
        binding.machine7.setOnClickListener { showInterstitialAndNavigate(Machine7::class.java) }
        binding.machine8.setOnClickListener { showInterstitialAndNavigate(Machine8::class.java) }
        binding.machine9.setOnClickListener { showInterstitialAndNavigate(Machine9::class.java) }
        binding.machine10.setOnClickListener { showInterstitialAndNavigate(Machine10::class.java) }
        binding.machine11.setOnClickListener { showInterstitialAndNavigate(Machine11::class.java) }
        binding.machine12.setOnClickListener { showInterstitialAndNavigate(Machine12::class.java) }
        binding.machine13.setOnClickListener { showInterstitialAndNavigate(Machine13::class.java) }
    }

    private fun showInterstitialAndNavigate(activity: Class<*>) {
        MyApplication.showInterstitialAdTrimmer(this, object : FullScreenAdListener() {
            override fun gotoNext() {
                super.gotoNext()
                startActivity(Intent(this@ChangeTrimmerActivity, activity))
            }
        }, "btn_back_change_trimmer")
    }

    private fun loadCollapsibleBanner() {
        val adId = if (AdConstants.TEST_ADS) {
            getString(R.string.BannerAd)
        } else {
            RemoteConfig.getString(RemoteConfig.CHANGE_TRIMMER_BANNER_AD_ID)
        }
        val makeCollapsible =
            RemoteConfig.getBoolean(RemoteConfig.CHANGE_TRIMMER_BANNER_MAKE_COLLAPSIBLE)
        val enable = RemoteConfig.getBoolean(RemoteConfig.ENABLE_CHANGE_TRIMMER_BANNER_AD)

        if (paymentSubscription.isPurchased.not() && internetConnection(this) && enable) {
            BannerAd.loadTop(binding.adaptivemain, adId.trim(), makeCollapsible)
        } else {
            binding.adaptivemain.visibility = View.GONE
        }
    }

    override fun handleBackPressed() {
        MyApplication.showInterstitialAdTrimmer(this, object : FullScreenAdListener() {
            override fun gotoNext() {
                super.gotoNext()
                finish()
            }
        }, "btn_back_change_trimmer")
    }

    private fun setVolumeMax() {
        audioManager.setStreamVolume(AudioManager.STREAM_MUSIC, 1, 0)
    }
}